include/fmt/format.h

```diff
@@ -3484,9 +3484,19 @@ template <typename Char> struct udl_arg {
   }
 };
 
+// Converts string literals to basic_string_view.
 template <typename Char, size_t N>
-FMT_CONSTEXPR basic_string_view<Char> literal_to_view(const Char (&s)[N]) {
-  return {s, N - 1};
+FMT_CONSTEXPR basic_string_view<Char> compile_string_to_view(
+    const Char (&s)[N]) {
+  // Remove trailing null character if needed. Won't be present if this is used
+  // with raw character array (i.e. not defined as a string).
+  return {s, N - (std::char_traits<Char>::to_int_type(s[N - 1]) == 0 ? 1 : 0)};
+}
+
+// Converts string_view-like types to basic_string_view.
+template <typename Char, typename String>
+FMT_CONSTEXPR basic_string_view<Char> compile_string_to_view(const String& s) {
```

Comment:
As commented earlier, please replace `String` with `std_string_view`.

Reply:
You've convinced me. I found workarounds for our internal code as well to not use folly::StringPiece, so I'm fine with making this more narrow.